app/org/sagebionetworks/bridge/hibernate/HibernateAccountDao.java

```diff
@@ -0,0 +1,490 @@
+            passwordHash = passwordAlgorithm.generateHash(password);
+        } catch (InvalidKeyException | InvalidKeySpecException | NoSuchAlgorithmException ex) {
+            throw new BridgeServiceException("Error creating password: " + ex.getMessage(), ex);
+        }
+        account.setPasswordAlgorithm(passwordAlgorithm);
+        account.setPasswordHash(passwordHash);
+
+        // Generate health code.
+        HealthId healthId = healthCodeService.createMapping(study.getStudyIdentifier());
+        account.setHealthId(healthId);
+
+        return account;
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public String createAccount(Study study, Account account, boolean sendVerifyEmail) {
+        // Initial creation of account. Fill in basic initial parameters.
+        String accountId = BridgeUtils.generateGuid();
+
+        HibernateAccount hibernateAccount = marshallAccount(account);
+        hibernateAccount.setId(accountId);
+        hibernateAccount.setStudyId(study.getIdentifier());
+        hibernateAccount.setCreatedOn(DateUtils.getCurrentMillisFromEpoch());
+        hibernateAccount.setModifiedOn(DateUtils.getCurrentMillisFromEpoch());
+        hibernateAccount.setPasswordModifiedOn(DateUtils.getCurrentMillisFromEpoch());
+        hibernateAccount.setStatus(sendVerifyEmail ? AccountStatus.UNVERIFIED : AccountStatus.ENABLED);
+
+        // Create account
+        try {
+            hibernateHelper.create(hibernateAccount);
+        } catch (ConcurrentModificationException ex) {
+            // account exists, but we don't have the userId, load the account
+            HibernateAccount otherAccount = getHibernateAccountByEmail(study, account.getEmail());
+            if (otherAccount != null) {
+                throw new EntityAlreadyExistsException(Account.class, "userId", otherAccount.getId());
+            } else {
+                throw new BridgeServiceException("Conflict creating an account, but can't find an existing " +
+                        "account with the same study and email");
+            }
+        }
+
+        // TODO send verify email
+
+        return accountId;
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public void updateAccount(Account account) {
+        String accountId = account.getId();
+        HibernateAccount accountToUpdate = marshallAccount(account);
+
+        // Can't change study, email, createdOn, or passwordModifiedOn.
+        HibernateAccount persistedAccount = hibernateHelper.getById(HibernateAccount.class, accountId);
+        if (persistedAccount == null) {
+            throw new EntityNotFoundException(Account.class, "Account " + accountId + " not found");
+        }
+        accountToUpdate.setStudyId(persistedAccount.getStudyId());
+        accountToUpdate.setEmail(persistedAccount.getEmail());
+        accountToUpdate.setCreatedOn(persistedAccount.getCreatedOn());
+        accountToUpdate.setPasswordModifiedOn(persistedAccount.getPasswordModifiedOn());
```

Comment:
The only place that does this is ParticipantService, and we already have logic to selectively marshall over the editable attributes.